fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java

```diff
@@ -119,6 +120,12 @@ protected FedoraResource doForward(final Resource resource) {
                 final boolean metadata = values.containsKey("path")
                         && values.get("path").endsWith("/" + FCR_METADATA);
 
+                final boolean timeMap = values.containsKey("path")
+                        && values.get("path").endsWith("/" + FCR_VERSIONS);
+                if (timeMap) {
+                    return new FedoraTimeMapImpl(node);
```

Comment:
Instead of the logic of determining if the `resource` is a TimeMap here, I believe it should happen in:
- https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java#L296 and
- https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/identifiers/NodeResourceConverter.java#L49

Reply:
@awoods I was thinking about the incoming path is /path/to/resource/fcr:versions. So the  JCR node returned from HttpResourceConverter.java#L120 
 (https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java#L120) is the JCR node for the resource and we can't determine whether we want the TimeMap or the resource in NodeResourceConverter.java#L49 (https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/identifiers/NodeResourceConverter.java#L49).

Do I miss anything? I think we can move the signature FedoraTimeMap.getTimeMap() back to FedoraResource.getTimeMap() if you think it works better. Thoughts?

Reply:
I think (and these Converters still confuse me) that we probably need the `getTimeMap()` in the `FedoraResourceImpl` and requests will have to be chained as `fcr:versions` is not an actual path.
So a request for `.../resource1/fcr:versions` gets `resource1` and then returns the result of the `getTimeMap()` function on that resource.
Which means that a request for `.../resource1/fcr:versions/1234-5678` will have to do the above and then find the appropriate child of the TimeMap.
This makes sense to me, but I'm not sure if this is a bad pattern.

Reply:
I agree, @whikloj . The question that remains for me is how much of the translation work we will be doing in [FedoraVersioning](https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraVersioning.java) and [FedoraVersions](https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraVersions.java), and how much in the translation chain.

Reply:
@awoods @whikloj: Basing on our discussions above, I've moved signature getTimeMap() back to FedoraResource/FedoraResourceImpl. See https://github.com/fcrepo4/fcrepo4/pull/1260/commits/79502171a145a19bb6e50b650a7452108a82405a.

Reply:
I do not think the suffix `/fcr:versions` should ever find its way into this method... because it should be intercepted in either FedoraVersioning or FedoraVersions.